### DISCUSSION TITLE
SWC support for astrojs @astrojs/react-swc

### DISCUSSION THREAD

**Discussion by ahqsoftwares** [withastro/roadmap#1073] [Proposal]
### Body

# Summary

Add support for SWC React, just like how we have @astrojs/react that will use @vitejs/plugin-react-swc under the hood

# Background & Motivation

It is important because on a codebase that has like 14000+ files. Babel itself seems to be the bottleneck.

**Comment by ematipico** [comment]
What's stopping you from publishing the integration yourself?

**Reply by ahqsoftwares** [reply]
I believe such a feature should be integrated into the astrojs ecosystem itself, just like how vite has SWC support out of the box via its plugins

**Reply by ahqsoftwares** [reply]
If you mean a PR to withastro/astro, sure, I shall make one

**Reply by ematipico** [reply]
No, I mean a package maintained by you. Many packages could be part of the org, but who's going to maintain them?

**Reply by ahqsoftwares** [reply]
okay, got it ✅
